hub/core/meta/dataset_meta.py

```diff
@@ -1,13 +1,21 @@
-from hub.constants import META_ENCODING
-import json
-
-from hub.core.typing import StorageProvider
+from hub.util.exceptions import MetaInvalidInitFunctionCall
+from hub.util.callbacks import CallbackList
+from hub.core.storage.provider import StorageProvider
+from hub.core.meta.meta import Meta
 from hub.util.keys import get_dataset_meta_key
 
 
-def write_dataset_meta(storage: StorageProvider, meta: dict):
-    storage[get_dataset_meta_key()] = json.dumps(meta).encode(META_ENCODING)
+class DatasetMeta(Meta):
+    tensors: CallbackList
```

Comment:
each `Meta` subclass has these properties at the top, this is to make intellisense pick up on the properties